six/seven_ans.py

```diff
@@ -0,0 +1,11 @@
+phone_dict = {'1':[''], '2':['A', 'B', 'C'], '3':['D', 'E', 'F'], '4':['G', 'H', 'I'], '5':['J', 'K', 'L'], '6':['M', 'N', 'O'], '7':['P', 'Q', 'R', 'S'], '8':['T', 'U', 'V'], '9':['W', 'X', 'Y', 'Z'], '0':['']}
+
+def nieoh_num_to_char(num):
+    if len(num)==1:
+        return phone_dict[num[0]]
+    ret = []
+    num = list(num)
```

Comment:
`list(x)` returns a new list by copying `x` so this doesn't actually save any memory. You could consider writing a wrapper which takes a list as input and then calling it from the main function. Otherwise I'd just omit this line.